### DISCUSSION TITLE
JSON-logging compliant with ECS-format

### DISCUSSION THREAD

**Discussion by johkin** [rabbitmq/rabbitmq-server#12433] [Ideas] [answered]
### Is your feature request related to a problem? Please describe.

We are ingesting our RabbitMQ-logs as json into an ELK-installation, but according to the ECS-rules (or perhaps Elasticsearch) we are having trouble to parse the produced json.
The object "mfa" is an array with mixed contents with both String and Integer/Long.

### Describe the solution you'd like

We would either like to see the mfa-object as a String-array or even better if one could specify to use the ECS standard in the generated json logs?

### Describe alternatives you've considered

There are other tools that can modify the logs once they are produced, but it would easier to produce schema-compliant logs to start with.

**Comment by johkin** [comment]
An example snippet of the logs: 
`"mfa": ["ra_server_sup_sup", "delete_server_rpc", 2]`

**Comment by ansd** [comment, marked as answer]
FWIW, https://github.com/rabbitmq/rabbitmq-server/pull/2927#issue-840999876 describes the different configuration options what fields to emit in the JSON log. You should be able to omit field `mfa` (module function arguments) completely (if that's acceptable for you).

**Reply by michaelklishin** [reply]
https://github.com/rabbitmq/rabbitmq-website/pull/2092

**Comment by johkin** [comment]
Thanks, that could be a good starting point! I'll look into it!

**Comment by michaelklishin** [comment]
@johkin we will not guess what specifically is logged in your case. Please provide a complete JSON example.

Given that #2927 exists and seeks to address the same fundamental problem, I'm moving this to a discussion.

**Comment by johkin** [comment]
The suggestion by @ansd work perfectly, thank you very much!
Would be nice to update the documentation around this, it's a powerful feature.

**Reply by michaelklishin** [reply]
https://github.com/rabbitmq/rabbitmq-website/issues/2089

**Reply by michaelklishin** [reply]
https://github.com/rabbitmq/rabbitmq-website/pull/2092
